### PR TITLE
[Feature](job)support cancel task and fix log invalid

### PR DESCRIPTION
**Pull request by CalvinKirs**
## Proposed changes

- Running task can be show and fix cancel fail
- When the insert task scheduling cycle is reached, if there are still tasks running, the scheduling of this task will be canceled at this time.
- refactor job status changes SQL
- Fix timer job window error
- Support cancel task

## todo
update other docs

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by CalvinKirs** [comment]
run buildall

**Comment by CalvinKirs** [comment]
run buildall

**Comment by CalvinKirs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/job/base/AbstractJob.java [hunk 1]
```diff
@@ -107,6 +110,27 @@ public void cancelTaskById(long taskId) throws JobException {
...
+        List<T> tasks = new ArrayList<>();
+        if (CollectionUtils.isEmpty(runningTasks)) {
+            return queryTasks();
+        }
+
+        List<T> historyTasks = queryTasks();
+        if (CollectionUtils.isNotEmpty(historyTasks)) {
+            tasks.addAll(historyTasks);
+        }
+        Map<Long, T> loadTask = tasks.stream().collect(Collectors.toMap(AbstractTask::getTaskId, a -> a));
```
[the remark this answers is not in the document]
Comment (CalvinKirs) [reply fe/fe-core/src/main/java/org/apache/doris/job/base/AbstractJob.java:123]: the key is  an index,

File: fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertTask.java [hunk 2]
```diff
@@ -148,7 +153,7 @@ public void cancel() throws JobException {
     @Override
     public List<String> getShowInfo() {
         if (null == loadJob) {
```
[the remark this answers is not in the document]
Comment (CalvinKirs) [reply fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertTask.java:155]: loadjob can only be obtained when execution is completed

File: regression-test/suites/job_p0/test_base_insert_job.groovy [hunk 3]
```diff
@@ -97,7 +97,7 @@ suite("test_base_insert_job") {
         assert true
     }
     sql """
-        STOP JOB for test_error_starts
+        STOP JOB where jobname =  'test_error_starts'
```
[the remark this answers is not in the document]
Comment (CalvinKirs) [reply regression-test/suites/job_p0/test_base_insert_job.groovy:100]: This is a bit difficult. Only long-running tasks can actually trigger the logical execution of cancel.

**Review by zddr — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
> Need to update the doc and add some test cases↳

The document is not yet public, I will provide it separately (when the show job is completed)

**Comment by CalvinKirs** [comment]
run buildall

**Comment by CalvinKirs** [comment]
run buildall

**Comment by CalvinKirs** [comment]
run clickbench-new (clickbench)

**Comment by CalvinKirs** [comment]
run clickbench

**Comment by CalvinKirs** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
run P0 Regression

**Comment by CalvinKirs** [comment]
run p0

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**